lib/http/middleware.go

```diff
@@ -183,7 +183,7 @@ func MiddlewareCORS(allowOrigin string) Middleware {
 
 			if allowOrigin != "" {
 				w.Header().Add("Access-Control-Allow-Origin", allowOrigin)
-				w.Header().Add("Access-Control-Allow-Headers", "authorization, Content-Type")
+				w.Header().Add("Access-Control-Allow-Headers", "Authorization, Content-Type, Depth, Destination, If, Lock-Token, Overwrite, TimeOut, Translate")
```

Comment:
Can you put this in the `serve webdav` middleware please? We don't need it to apply to all of rclone's http server.

Likewise the webdav only `Access-Control-Allow-Methods` should be in there too.

I'm not sure exactly how to do that - it might require a bit of plumbing in rclone.

Reply:
> I'm not sure exactly how to do that - it might require a bit of plumbing in rclone.

This is the dreadful part. I'm not really a Go developer and it may take me quite a while to figure that out. :(
I agree with that comment. We can also separate these issues. This above is a quick fix for allowing headers, which fixes the WebDAV functionality. Finding the correct way to limit it to `webdav serve` will take longer, it seems. I looked at the `serve webdav` code and do not understand much there, as I've mentioned in the related issue.